### ISSUE TITLE
A question during training

### ISSUE THREAD

**Issue by chu2001dd-dot** [adrienpetralia/NILMFormer#7]
2025-10-29 21:48:30,871 - INFO - ---- Run experiments with provided parameters ---- 
2025-10-29 21:48:30,872 - INFO - Dataset: UKDALE 
2025-10-29 21:48:30,872 - INFO - Sampling Rate: 1min 
2025-10-29 21:48:30,873 - INFO - Window Size: 128 
2025-10-29 21:48:30,873 - INFO - Appliance : Kettle 
2025-10-29 21:48:30,873 - INFO - Model: NILMFormer 
2025-10-29 21:48:30,873 - INFO - Seed: 0 
2025-10-29 21:48:30,873 - INFO - -------------------------------------------------- 
2025-10-29 21:48:30,873 - INFO - Process data ... 
2025-10-29 21:48:58,232 - INFO - ... Done. 
2025-10-29 21:48:58,388 - INFO - Model training... 
2025-10-29 21:49:22,276 - INFO - Epoch [1/50] 
2025-10-29 21:49:22,276 - INFO - Train loss : 0.0003 
2025-10-29 21:49:22,276 - INFO - Valid loss : 0.0002 
2025-10-29 21:49:46,348 - INFO - Epoch [2/50] 
2025-10-29 21:49:46,348 - INFO - Train loss : 0.0002 
2025-10-29 21:49:46,348 - INFO - Valid loss : 0.0002 
2025-10-29 21:50:10,090 - INFO - Epoch [3/50] 
2025-10-29 21:50:10,091 - INFO - Train loss : 0.0002 
2025-10-29 21:50:10,091 - INFO - Valid loss : 0.0002 
2025-10-29 21:50:33,600 - INFO - Epoch [4/50] 
2025-10-29 21:50:33,601 - INFO - Train loss : 0.0002 
2025-10-29 21:50:33,601 - INFO - Valid loss : 0.0002 
2025-10-29 21:50:57,737 - INFO - Epoch [5/50] 
2025-10-29 21:50:57,737 - INFO - Train loss : 0.0002 
2025-10-29 21:50:57,737 - INFO - Valid loss : 0.0002 
2025-10-29 21:51:22,850 - INFO - Epoch [6/50] 
2025-10-29 21:51:22,851 - INFO - Train loss : 0.0001 
2025-10-29 21:51:22,851 - INFO - Valid loss : 0.0002 
2025-10-29 21:51:48,043 - INFO - Epoch [7/50] 
2025-10-29 21:51:48,043 - INFO - Train loss : 0.0001 
2025-10-29 21:51:48,043 - INFO - Valid loss : 0.0002 
2025-10-29 21:52:12,527 - INFO - Epoch [8/50] 
2025-10-29 21:52:12,527 - INFO - Train loss : 0.0001 
2025-10-29 21:52:12,528 - INFO - Valid loss : 0.0001 
2025-10-29 21:52:37,778 - INFO - Epoch [9/50] 
2025-10-29 21:52:37,778 - INFO - Train loss : 0.0001 
2025-10-29 21:52:37,778 - INFO - Valid loss : 0.0002 
2025-10-29 21:53:04,750 - INFO - Epoch [10/50] 
2025-10-29 21:53:04,750 - INFO - Train loss : 0.0001 
2025-10-29 21:53:04,750 - INFO - Valid loss : 0.0002 
2025-10-29 21:53:31,008 - INFO - Epoch [11/50] 
2025-10-29 21:53:31,008 - INFO - Train loss : 0.0001 
2025-10-29 21:53:31,008 - INFO - Valid loss : 0.0002 
2025-10-29 21:53:58,043 - INFO - Epoch [12/50] 
2025-10-29 21:53:58,043 - INFO - Train loss : 0.0001 
2025-10-29 21:53:58,043 - INFO - Valid loss : 0.0002 
2025-10-29 21:54:22,561 - INFO - Epoch [13/50] 
2025-10-29 21:54:22,561 - INFO - Train loss : 0.0001 
2025-10-29 21:54:22,561 - INFO - Valid loss : 0.0001 
2025-10-29 21:54:46,976 - INFO - Epoch [14/50] 
2025-10-29 21:54:46,976 - INFO - Train loss : 0.0001 
2025-10-29 21:54:46,976 - INFO - Valid loss : 0.0001 
2025-10-29 21:55:12,408 - INFO - Epoch [15/50] 
2025-10-29 21:55:12,409 - INFO - Train loss : 0.0001 
2025-10-29 21:55:12,409 - INFO - Valid loss : 0.0001 
2025-10-29 21:55:38,112 - INFO - Epoch [16/50] 
2025-10-29 21:55:38,112 - INFO - Train loss : 0.0001 
2025-10-29 21:55:38,112 - INFO - Valid loss : 0.0001

I only truncated a portion of the code， Train loss and Valid loss，are their values normal?
Will only one. pt file be generated in the end? How should I proceed to compare the results with those in allresults?

I hope you can solve my problem amidst your busy schedule, thank you!!!

**Comment by xailxail** [comment]
where are you from
?

**Comment by xailxail** [comment]
I can help solve this problem

**Comment by adrienpetralia** [comment]
Hey,

### 1. Loss values

Yes, those values are perfectly normal. UKDALE data are of small magnitudes and if you used the MaxScaler in the dataset, the MSE/MAE will naturally show up around 1e-4.

Neertheless, to make this easier to read when training, I just updated the trainer to print 6 decimal places for the losses, so you can better track small improvements.

### 2. About the .pt checkpoint (comparison with metrics in the csv file)

- A checkpoint is saved only if save_checkpoint=True.
- What’s inside the checkpoint:
  1. model_state_dict (handles DataParallel via model.module.state_dict()),
  2. optimizer_state_dict,
  3. loss_train_history, loss_valid_history,
  4. value_best_loss, epoch_best_loss, time_best_loss
- During evaluate(...), if save_checkpoint=True, it will save again (same file), augmenting the log with evaluation metrics (e.g., test_metrics_timestamp, test_metrics_win, timings). So if you want to compare your results with the ones provided in the csv file in this repo, you just have to extract the *test_metrics_timestamp* dict (**NB:** don't forget that the results obtain in our paper are average on at least 5 runs with different seed; i.e., train/valid/test splits).

Cheers,
Adrien
